packages/devextreme/testing/tests/DevExpress.ui.widgets/menu.tests.js

```diff
@@ -380,6 +384,82 @@ QUnit.module('Menu rendering', {
         fixtures.simple.drop();
     });
+
+    QUnit.test('submenu should init ScrollView', function(assert) {
+        const menu = createMenu({
+            items: [{
+                text: 'item 1',
+                items: [{
+                    text: 'item 11',
+                }],
+            }],
+            showFirstSubmenuMode: 'onClick'
+        });
+        const $item1 = $(menu.element).find(`.${DX_MENU_ITEM_CLASS}`).eq(0);
+
+        $($item1).trigger('dxclick');
+        this.clock.tick(0);
+
+        const $submenu = $(`.${DX_SUBMENU_CLASS}`);
+
+        assert.strictEqual($submenu.length, 1, 'only 1 submenu exists');
+        assert.ok($submenu.hasClass(DX_SCROLLVIEW_CLASS), 'ScrollView initialized');
+    });
+
+    QUnit.test('ScrollView should be initialised on 2nd level submenu', function(assert) {
+        if(!isDeviceDesktop(assert)) {
+            return;
+        }
+
+        const menu = createMenu({
+            items: [{
+                text: 'item 1',
+                items: [{
+                    text: 'item 11',
+                    items: [{
+                        text: 'item 111',
+                    }],
+                }],
+            }],
+            showFirstSubmenuMode: 'onClick',
+            showSubmenuMode: { name: 'onHover', delay: 0 },
+        });
+        const $item1 = $(menu.element).find(`.${DX_MENU_ITEM_CLASS}`).eq(0);
+
+        $($item1).trigger('dxclick');
+        this.clock.tick(0);
+
+        const submenu = getSubMenuInstance($item1);
+        this.clock.tick(MOUSETIMEOUT);
+        assert.ok(submenu._overlay.option('visible'));
+
+        const $menuItem = $($(submenu._overlay.content()).find(`.${DX_MENU_ITEM_CLASS}`).first());
+        $(submenu.itemsContainer()).trigger($.Event('dxhoverstart', { target: $menuItem.get(0) }));
+        $($menuItem).trigger('dxpointermove');
+        this.clock.tick(0);
+        const $submenu = $($(submenu._overlay.content()).find(`.${DX_SUBMENU_CLASS}`).eq(1));
+        assert.ok($submenu.hasClass(DX_SCROLLVIEW_CLASS), 'ScrollView initialized on nested submenu');
+    });
+
+    QUnit.test('height of the submenu should be limited', function(assert) {
+        const menu = createMenu({
+            items: [{
+                text: 'item 1',
+                items: (new Array(99)).fill(null).map((_, idx) => ({ text: idx })),
+            }],
+            showFirstSubmenuMode: 'onClick',
+        });
+
+        const $item1 = $(menu.element).find(`.${DX_MENU_ITEM_CLASS}`).eq(0);
+
+        $($item1).trigger('dxclick');
+        this.clock.tick(0);
+
+        const $submenu = $(`.${DX_SUBMENU_CLASS}`);
+
+        assert.ok($submenu.find(`.${DX_SCROLLVIEW_CONTENT_CLASS}`).height() > $(window).height(), 'total height of submenu exceeds the window height');
+        assert.strictEqual($submenu.outerHeight(), $(window).height(), 'menu uses the full height of the window');
```

Comment:
hm, why should it take the whole height of the window?
I believe it should only take the place below the clicked item.

Also i think scenario for 2nd level submenu should be tested as well.